### PR TITLE
Add SignedEvmMapping type

### PR DESCRIPTION
**Pull request by saibatizoku**
Adds support for signing `EvmMapping`s, and verifying that the signing key corresponds to the `evm_address` field.

### CODE REVIEW

File: chain-impl-mockchain/src/certificate/evm_mapping.rs [hunk 1]
```diff
@@ -127,6 +163,101 @@ impl DeserializeFromSlice for EvmMapping {
...
+    fn rlp_append(&self, s: &mut RlpStream) {
+        s.begin_list(2);
+        s.append(&U256::from_big_endian(self.account_id.as_ref().as_ref()));
+        s.append(&self.evm_address);
+    }
+}
+
+/// Represents a signed mapping between a Jormungandr account and an EVM account.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct SignedEvmMapping {
```
Comment (Mr-Leshiy) [inline chain-impl-mockchain/src/certificate/evm_mapping.rs:197]: Why is it needed to create a separate struct ? Do we really need signed and unsigned types of the `EvmMapping` transaction ? I suppose that we can leave only `SignedEvmMapping`.

File: chain-impl-mockchain/src/certificate/evm_mapping.rs [hunk 2]
```diff
@@ -127,6 +166,123 @@ impl DeserializeFromSlice for EvmMapping {
...
```
Comment (Mr-Leshiy) [inline chain-impl-mockchain/src/certificate/evm_mapping.rs:200]: It is better to reuse original `EvmMapping` for the `account_id` and `evm_address` fields

Comment (Mr-Leshiy) [inline chain-impl-mockchain/src/certificate/evm_mapping.rs:200]: We also should replace the current usage of the `EvmMapping` type with the `SIgnedEvmMapping` inside the certificate

File: chain-impl-mockchain/src/certificate/evm_mapping.rs [hunk 3]
```diff
@@ -127,6 +166,123 @@ impl DeserializeFromSlice for EvmMapping {
...
+            account_id: other.account_id.clone(),
+            evm_address: other.evm_address,
+        }
+    }
+}
+
+#[cfg(feature = "evm")]
+impl SignedEvmMapping {
+    /// Verifies that the signing key corresponds to the `evm_address`.
+    pub fn verify(&self) -> Result<bool, Error> {
```
Comment (Mr-Leshiy) [inline chain-impl-mockchain/src/certificate/evm_mapping.rs:226]: We should use this validation inside `apply_map_accounts` fn (or any other place, but during the transaction processing inside the ledger)

**Review by Mr-Leshiy — CHANGES_REQUESTED** [review]

File: chain-impl-mockchain/src/certificate/evm_mapping.rs (hunk 2 above)
Comment (saibatizoku) [reply chain-impl-mockchain/src/certificate/evm_mapping.rs:121]: This is done.

**Comment by saibatizoku** [comment]
closing this PR in favor of https://github.com/input-output-hk/chain-libs/pull/832

**Pull request closed without merging**